tests/nativeapp/codegen/snowpark/test_python_processor.py

```diff
@@ -286,3 +288,147 @@ def test_execute_in_sandbox_none_entity_case2(mock_jinja, mock_sandbox):
         py_file="some_file", deploy_root=Path("some/path"), kwargs={}
     )
     assert entity is None
+
+
+def test_generate_create_sql_ddl_statements_w_all_entries(
+    native_app_codegen_instance, snapshot
+):
+    assert generate_create_sql_ddl_statements(native_app_codegen_instance) == snapshot
+
+
+def test_generate_create_sql_ddl_statements_none():
+    ex_fn = {
+        "object_type": "PROCEDURE",
+        "object_name": "CORE.MYFUNC",
+        "anonymous": True,
+    }
+    assert generate_create_sql_ddl_statements(ex_fn=ex_fn) is None
+
+
+def test_generate_create_sql_ddl_statements_w_select_entries(
+    native_app_codegen_instance, snapshot
+):
+    native_app_codegen_instance["replace"] = False
+    native_app_codegen_instance["all_imports"] = ""
+    native_app_codegen_instance["all_packages"] = ""
+    native_app_codegen_instance["external_access_integrations"] = None
+    native_app_codegen_instance["secrets"] = None
+    native_app_codegen_instance["execute_as"] = None
+    native_app_codegen_instance["inline_python_code"] = None
+    assert generate_create_sql_ddl_statements(native_app_codegen_instance) == snapshot
+
+
+def test_generate_grant_sql_ddl_statements_none():
+    ex_fn = {"application_roles": None}
+    assert generate_grant_sql_ddl_statements(ex_fn=ex_fn) is None
+    ex_fn["application_roles"] = []
+    assert generate_grant_sql_ddl_statements(ex_fn=ex_fn) is None
+
+
+def test_generate_grant_sql_ddl_statements():
+    ex_fn = {
+        "object_type": "TABLE_FUNCTION",
+        "object_name": "CORE.MYFUNC",
+        "application_roles": ["APP_ADMIN", "APP_VIEWER"],
+    }
+    actual = generate_grant_sql_ddl_statements(ex_fn=ex_fn)
+    expected = "GRANT USAGE ON TABLE FUNCTION CORE.MYFUNC\nTO APPLICATION ROLE APP_ADMIN;\n\nGRANT USAGE ON TABLE FUNCTION CORE.MYFUNC\nTO APPLICATION ROLE APP_VIEWER;\n"
+    assert actual == expected
+
+
+def test_generate_sql_ddl_statements_empty(native_app_project_instance):
+    dir_structure = {
+        "a/b/c/main.py": "# this is a file\n",
+        "a/b/c/data.py": "# this is a file\n",
+        "output/deploy": None,
```

Comment:
The utility function takes files, not directories. This looks out of place, I _think_ it will try to create an empty file named `deploy`